src/PKSim.Infrastructure/ORM/Repositories/ContainerParametersNotCommonForAllSpeciesRepository.cs

```diff
@@ -0,0 +1,78 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+using OSPSuite.Core.Domain;
+using OSPSuite.Utility.Collections;
+using PKSim.Core.Model;
+using PKSim.Core.Repositories;
+
+namespace PKSim.Infrastructure.ORM.Repositories
+{
+   public class ContainerParametersNotCommonForAllSpeciesRepository: StartableRepository<ContainerParameterBySpecies>, IContainerParametersNotCommonForAllSpeciesRepository
+   {
+      private readonly IFlatContainerRepository _flatContainerRepository;
+      private readonly IFlatContainerParametersNotCommonForAllSpeciesRepository _flatContainerParametersNotCommonForAllSpeciesRepository;
+
+      private readonly List<ContainerParameterBySpecies> _containerParametersNotCommonForAllSpecies;
+      private readonly ICache<string, List<string>> _parametersNotCommonForAllSpeciesByContainer;
+      private readonly ICache<string, string> _parametersNotCommonForAllSpeciesByFullPath;
+
+      public ContainerParametersNotCommonForAllSpeciesRepository(
+         IFlatContainerRepository flatContainerRepository,
+         IFlatContainerParametersNotCommonForAllSpeciesRepository flatContainerParametersNotCommonForAllSpeciesRepository)
+      {
+         _flatContainerRepository = flatContainerRepository;
+         _flatContainerParametersNotCommonForAllSpeciesRepository = flatContainerParametersNotCommonForAllSpeciesRepository;
+
+         _containerParametersNotCommonForAllSpecies = new List<ContainerParameterBySpecies>();
+         _parametersNotCommonForAllSpeciesByContainer = new Cache<string, List<string>>();
+         _parametersNotCommonForAllSpeciesByFullPath = new Cache<string, string>();
+      }
+
+      protected override void DoStart()
+      {
+         var flatContainerParametersNotCommonForAllSpecies = _flatContainerParametersNotCommonForAllSpeciesRepository.All().ToList();
+
+         foreach(var containerParameter in flatContainerParametersNotCommonForAllSpecies)
+         {
+            var containerPath = _flatContainerRepository.ContainerPathFrom(containerParameter.ContainerId).ToString();
+            var containerParameterBySpecies = new ContainerParameterBySpecies()
+            {
+               ContainerPath = containerPath,
+               ParameterName = containerParameter.ParameterName,
+               SpeciesCount = containerParameter.SpeciesCount
+            };
+            _containerParametersNotCommonForAllSpecies.Add(containerParameterBySpecies);
+
+            //cache by full path
+            var fullPath = $"{containerPath}{ObjectPath.PATH_DELIMITER}{containerParameter.ParameterName}";
```

Comment:
do we have a function somewhere for concatenation of path `strings`?
We can add/insert a string to an object path **object**, but I didn't find a function like `CombinePath(string,string)`